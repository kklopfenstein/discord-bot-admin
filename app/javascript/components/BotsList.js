import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Paper from '@mui/material/Paper';

export default () => {
  const [bots, setBots] = useState([]);

  let loadBots = () => {
    axios.get('/api/v1/bots')
      .then(function (response) {
        let bots = []
        response.data.forEach((bot) => {
          const newBot = {
            id: bot.id,
            name: bot.name,
            token: bot.token
          };

          bots.push(newBot);
        });
        setBots(bots);
      })
      .catch(function(error) {
        console.error(error);
      });
  };

  useEffect(() => {
    loadBots();
  }, []);

  return (
    <>
      <TableContainer component={Paper}>
        <Table sx={{ minWidth: 650 }}>
          <TableHead>
            <TableRow>
              <TableCell>Bot name</TableCell>
              <TableCell>Bot token</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {bots.map((bot) => (
              <TableRow key={bot.id}>
                <TableCell>{bot.name}</TableCell>
                <TableCell>{bot.token}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    <div>
      <Link to="/bots/new">New Bot</Link>
    </div>
   </>
  )
}