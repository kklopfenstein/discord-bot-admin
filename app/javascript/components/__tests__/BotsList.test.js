import React from 'react';
import renderer, { act } from 'react-test-renderer';
import BotsList from '../BotsList';
import mockAxios from 'jest-mock-axios';

jest.mock('react-router-dom', () => ({
  Link: () => {
    const MockName = "mock-name";
    return <MockName />;
  }
}));


afterEach(() => {
  mockAxios.reset();
});

describe('BotList', () => {
  it('displays bots', () => {
    let component;
    act(() => {
      component = renderer.create(
        <BotsList />
      );
    });

    expect(mockAxios.get).toHaveBeenCalledWith('/api/v1/bots');
    let responseObj = { data: [
      {
        id: 1,
        name: "testbot",
        token: "OTM532IwNzgyMDEzNDZhhhgw.Yd6C6Q.Pslx7BdYAlmCsTkJDndWHv8be3z"
      }
    ]};

    act(() => {
      mockAxios.mockResponse(responseObj);
    });

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays no bots', () => {
    let component;
    act(() => {
      component = renderer.create(
        <BotsList />
      );
    });

    expect(mockAxios.get).toHaveBeenCalledWith('/api/v1/bots');
    let responseObj = { data: [] };

    act(() => {
      mockAxios.mockResponse(responseObj);
    });

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});