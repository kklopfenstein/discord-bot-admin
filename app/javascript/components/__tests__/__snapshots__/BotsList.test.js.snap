// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BotList displays bots 1`] = `
Array [
  <div
    className="MuiBox-root css-1ebprri"
  >
    <mock-name />
  </div>,
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      className="MuiTable-root css-quj9j6-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Bot name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Bot token
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          />
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          />
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            testbot
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            OTM532IwNzgyMDEzNDZhhhgw.Yd6C6Q.Pslx7BdYAlmCsTkJDndWHv8be3z
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <mock-name />
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <mock-name />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;

exports[`BotList displays no bots 1`] = `
Array [
  <div
    className="MuiBox-root css-1ebprri"
  >
    <mock-name />
  </div>,
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      className="MuiTable-root css-quj9j6-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Bot name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Bot token
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          />
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          />
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      />
    </table>
  </div>,
]
`;
